v0/destinations/am/transform.js

```diff
@@ -263,11 +301,16 @@ function responseBuilderSimple(
             if (SpecedTraits.includes(trait)) {
               const mapping = TraitsMapping[trait];
               Object.keys(mapping).forEach(key => {
-                set(
-                  rawPayload,
-                  `user_properties.${key}`,
-                  get(traits, mapping[key])
-                );
+                const checkKey = get(rawPayload.user_properties, key);
+                // this is done only if we want to add default values under address to the user_properties
+                // these values are also sent to the destination at the top level.
+                if (!isDefinedAndNotNull(checkKey)) {
```

Comment:
I see that we are checking here to not override the existing values. Where are we setting them the first time?

Reply:
For example if `platform` is there in traits it didn't go to the top level for amplitude event. but went with user_properties. Now to exist with this logic we map it to the top level and also send it in the use_properties.